src/ui/features/xp-drop/XpDropClaimFlow/XpDropClaim.tsx

```diff
@@ -0,0 +1,526 @@
+          </UIText>
+        }
+        value={
+          <UIText
+            kind="headline/h2"
+            className={styles.zerionOgGradient}
+            style={{ wordBreak: 'break-all' }}
+          >
+            {formatXp(zerionOg)} XP
+          </UIText>
+        }
+      />
+      <StatsItem
+        title={
+          <UIText kind="small/accent" color="var(--neutral-700)">
+            Activity
+          </UIText>
+        }
+        value={
+          <UIText kind="headline/h2" style={{ wordBreak: 'break-all' }}>
+            {formatXp(activity)} XP
+          </UIText>
+        }
+      />
+    </HStack>
+  );
+}
+
+function HCenter({ children }: { children: React.ReactNode }) {
+  return (
+    <div style={{ display: 'flex', justifyContent: 'center' }}>{children}</div>
+  );
+}
+
+function XpLevel({ claimedXp, level }: { claimedXp: number; level: number }) {
+  return (
+    <Frame style={{ padding: '14px 16px' }}>
+      <HStack gap={16} alignItems="center">
+        <GradientBorder
+          borderColor="linear-gradient(90deg, #a024ef 0%, #fdbb6c 100%)"
+          borderWidth={4}
+          borderRadius={16}
+          backgroundColor="var(--white)"
+        >
+          <UIText kind="headline/hero" className={styles.levelGradient}>
+            {level}
+          </UIText>
+        </GradientBorder>
+        <UIText kind="small/accent" color="var(--neutral-700)">
+          {formatXp(claimedXp)} XP claimed!
+          <br />
+          Your wallet is now level {level}
+        </UIText>
+      </HStack>
+    </Frame>
+  );
+}
+
+export function XpDropClaim() {
+  useBackgroundKind({ kind: 'white' });
+  const navigate = useNavigate();
+
+  const { data: currentWallet, refetch: refetchCurrentWallet } = useQuery({
+    queryKey: ['wallet/uiGetCurrentWallet'],
+    queryFn: () => walletPort.request('uiGetCurrentWallet'),
+  });
+
+  const setCurrentAddressMutation = useMutation({
+    mutationFn: (address: string) => setCurrentAddress({ address }),
+    onSuccess() {
+      refetchCurrentWallet();
+    },
+  });
+
+  const [selectedWallet, setSelectedWallet] = useState(currentWallet);
+  const [claimedInfo, setClaimedInfo] = useState<{
+    address: string;
+    meta: WalletMeta | null;
+  } | null>(null);
+
+  const { data: walletGroups, ...walletGroupsQuery } = useWalletGroups();
+  const { signerWalletGroups, signerWalletAddresses } = useMemo(() => {
+    const signerWalletGroups =
+      walletGroups?.filter((group) =>
+        isSignerContainer(group.walletContainer)
+      ) ?? [];
+    const signerWalletAddresses = signerWalletGroups
+      .flatMap((group) => group.walletContainer.wallets)
+      .map((wallet) => wallet.address);
+    return { signerWalletGroups, signerWalletAddresses };
+  }, [walletGroups]);
+
+  const { data: walletsMeta, ...walletsMetaQuery } = useQuery({
+    queryKey: ['ZerionAPI.getWalletsMeta', signerWalletAddresses],
+    queryFn: () => getWalletsMetaByChunks(signerWalletAddresses),
```

Comment:
Must be called as a method of `ZerionAPI`?

Reply:
Thanks, same on Overview.tsx, will update in a minute

Reply:
Oh, I even have `useWalletsMetaByChunks` in this branch, I'll better use this hook to avoid mistakes like that